example/nextjs/lib/withNamespaces.js

```diff
@@ -0,0 +1,25 @@
+/*
+
+  Here we are just returning the built-in `withNamespaces`
+  HOC from `react-i18next` core, but with one added
+  piece of functionality: we keep track of namespaces
+  used by simply modifying an array on our i18n instance
+  directly.
+
+  We will use this nsFromReactTree array in SSR to
+  serialize and send initial translation data to the client
+  for exactly the namespaces we used.
+
+  This allows you to wrap pages, components, and anything
+  else with `withNamespaces`.
+
+*/
+
+import { withNamespaces } from 'react-i18next';
+
+import i18n from '../i18n';
+
+export default (namespaces = []) => {
+  i18n.nsFromReactTree = [...new Set(i18n.nsFromReactTree.concat(namespaces))];
+  return withNamespaces(namespaces);
```

Comment:
Does this HOC copy the `getInitialProps` method when used on top level pages? Otherwise getInitialProps won't work on pages wrapped in withNamespaces

Reply:
@timneutkens That's a great point.

The `withNamespaces` HOC from `react-i18next` is already using `hoist-non-react-statics`. I just tested this by adding a `getInitialProps` static to the `page-with-hoc` in the example, and can successfully pass props down.

@jamuhl and I were discussing, and we'd like to merge this functionality into the `react-i18next` core regardless.

Reply:
@isaachinman yes...absolutely...if you like to provide a PR to merge it in

Reply:
@isaachinman agree